# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/tar/no_same_owner.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/extract.c.orig	Mon Jun  3 14:56:04 2002
+++ ./src/extract.c	Mon Jun  3 14:56:57 2002
@@ -162,7 +162,7 @@
      extract as the original owner.  Or else, if we are running as a user,
      leave the owner and group as they are, so we extract as that user.  */
 
-  if (we_are_root || same_owner_option)
+  if (!no_same_owner_option && (we_are_root || same_owner_option))
     {
 #if HAVE_LCHOWN
 
--- ./src/common.h.orig	Mon Jun  3 14:57:49 2002
+++ ./src/common.h	Mon Jun  3 15:13:56 2002
@@ -196,6 +196,9 @@
 GLOBAL int no_recurse_option;
 
 /* Boolean value.  */
+GLOBAL int no_same_owner_option;
+
+/* Boolean value.  */
 GLOBAL int numeric_owner_option;
 
 /* Boolean value.  */
--- ./src/tar.c.orig	Mon Jun  3 15:13:28 2002
+++ ./src/tar.c	Mon Jun  3 15:15:53 2002
@@ -205,6 +205,7 @@
   {"newer-mtime", required_argument, NULL, NEWER_MTIME_OPTION},
   {"null", no_argument, NULL, NULL_OPTION},
   {"no-recursion", no_argument, NULL, NO_RECURSE_OPTION},
+  {"no-same-owner", no_argument, &no_same_owner_option, 1},
   {"numeric-owner", no_argument, &numeric_owner_option, 1},
   {"old-archive", no_argument, NULL, 'o'},
   {"one-file-system", no_argument, NULL, 'l'},
@@ -305,6 +306,7 @@
       --atime-preserve         don't change access times on dumped files\n\
   -m, --modification-time      don't extract file modified time\n\
       --same-owner             try extracting files with the same ownership\n\
+      --no-same-owner          don't extract files with the same ownership\n\
       --numeric-owner          always use numbers for user/group names\n\
   -p, --same-permissions       extract all protection information\n\
       --preserve-permissions   same as -p\n\
